brain/morphology.cpp

```diff
@@ -70,11 +87,35 @@ class Morphology::Impl
         return result;
     }
 
-    Vector4fs getSectionSamples(
-        const size_t sectionID, const floats& samplePoints ) const
+    float getSectionLength( const uint32_t sectionID ) const
+    {
+        if( _sectionLengths.size() <= sectionID )
+            _sectionLengths.resize( sectionID + 1 );
+
+        float& length = _sectionLengths[sectionID];
+
+        if( length == 0 && ( *types )[sectionID] != SECTION_SOMA )
+            length = _computeSectionLength( sectionID );
+        return length;
+    }
+
+    Vector4fs getSectionSamples( const uint32_t sectionID ) const
+    {
+        const SectionRange range = getSectionRange( sectionID );
+        Vector4fs result;
+        result.reserve( range.second - range.first );
+        for( size_t i = range.first; i != range.second; ++i )
```

Comment:
Do you really think any of the below is better?:

    result.insert( result.end(),
                   points->begin() + range.first, points->begin() + range.second)

    std::copy( points->begin() + range.first, points->begin() + range.second,
               std::back_inserter( result ));